common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj

```diff
@@ -48,39 +49,101 @@
   [concept-type fields]
   (map #(q2e/query-field->elastic-field (keyword %) concept-type) fields))
 
+(defn- query->execution-params
+  "Returns the Elasticsearch execution parameters extracted from the query. These are the
+  actual ES parameters as epxected by the elastisch library. The :scroll-id parameter is special
+  and is stripped out before calling elastisch to determine whether a normal search call or a
+  scroll call should be made."
+  [query]
+  (let [{:keys [page-size offset concept-type aggregations highlights scroll scroll-id]} query
```

Comment:
The `query` parameter passed in here is not a URL query or raw Lucene query - it was constructed programmatically in the CMR (not provided directly by a client and not through string concatenation) as a collection of clojure records with parameter validation, etc. Also, Elasticsearch is not like a SQL database where you have one endpoint and the SQL determines whether you are reading, writing, or deleting. ES provides a RESTful interface and we always send client queries to the `search` endpoint that only performs reads (via the elastisch library). To the best of my knowledge injection attacks are a non-issue.